src/main/java/com/linecorp/armeria/common/ServiceInvocationContext.java

```diff
@@ -149,6 +154,116 @@ public final EventLoop eventLoop() {
     }
 
     /**
+     * Returns a {@link EventExecutor} that will make sure this invocation is set
+     * as the current invocation before executing any callback. This should
+     * almost always be used for executing asynchronous callbacks in service
+     * code to make sure features that require the invocation context work
+     * properly. Most asynchronous libraries like
+     * {@link java.util.concurrent.CompletableFuture} provide methods that
+     * accept an {@link Executor} to run callbacks on.
+     */
+    public final EventExecutor contextAwareEventExecutor() {
+        return new ServiceInvocationContextAwareEventExecutor(this, eventLoop());
```

Comment:
How about renaming to `contextAwareEventLoop()` and returning `EventLoop`? Not really a very useful change, but a user will find that this method is related with `eventLoop()` method more easily.